### ISSUE TITLE
Trouble with binding handler to the paging navigators' "click" event on IE 8

### ISSUE THREAD

**Issue by bboyz269** [tabalinas/jsgrid#164]
Hi, It seems that I can't bind any handler on the navigator-link (those `<a href="javascript:void(0);">`s) on IE8. Others (IE9+, FF, Chrome, ...) work fine through. Code is very basic, with your demos\basic.html:
```Javascript
$(document).on("click", "a[href]", function() { alert("clicked"); });
```
Nothing is alerted when i click on the navigators.

What I actually want to archive is: On my page, I bound a handler to "beforeunload" event to prompt every time user try to leave the page. I don't know why but on IE8, click on the navigator link will also trigger "beforeunload" (Again, other browsers work fine, no dialog is promted). 
I'm trying to dirty fix this but it lead to the issue above. Code on my page is something like:
```Javascript
var navigatorClicked = false; // var to check if navigator link clicked
$(function() {
    // bind event
    $(window).on("beforeunload", function() {
        if (navigatorClicked) {
            navigatorClicked = false; // reset var
        } else {
            return "leaving ?";
        }
    });
    // dirty fix
    $(document).on("click", "a[href]", function() {
        navigatorClicked = true;
    });

    // create jsGrid
    /* ... */
});
```
Execute above code into your demo page (sorry for the Japanese interface):
![2015-12-09_162223](https://cloud.githubusercontent.com/assets/4453811/11679117/61e15528-9e91-11e5-8bb6-cd89d756260a.png)

Thank you for your help!

**Comment by tabalinas** [comment]
Are you sure this works for normal links outside of jsGrid?
Are you using a jQuery version supporting IE8?

**Comment by bboyz269** [comment]
Hi, thank you for your suggestion. The script is pretty straightforward and works on other pages of my site (just tested again). I'm using jquery-1-10-2, all 1.x versions support IE8 if I recalled correctly.

And one other thing, in my page, the "fix" actually works for the numbered-link. The only link that does not work is the elipsis-link (...) while no link works at all on your demo page (basic.html).

![2015-12-10_084045](https://cloud.githubusercontent.com/assets/4453811/11702619/761952e0-9f1b-11e5-88b7-a606af0af707.png)

Been googling around just to find more wierd stuff about IE8. I'll update if I find anything else.

**Comment by tabalinas** [comment]
Could you make a fork of the fiddle http://jsfiddle.net/tabalinas/zzzfmpgt/ modifying it, so I can reproduce the issue on my side?

**Comment by bboyz269** [comment]
Will do on Monday, thank you!

**Comment by bboyz269** [comment]
Hi, I've just update the fiddle here (http://jsfiddle.net/zzzfmpgt/15/) with:
* Attach click event on a[href]
* Reduce pageSize to 3 so [...] will be displayed.

Notes:
- You can get a copy of IE8 VM here https://dev.windows.com/en-us/microsoft-edge/tools/vms/windows/
- jsfiddle has problems rendering on IE8 so you should use the embedded fullscreen page http://jsfiddle.net/zzzfmpgt/15/embedded/result/

Thank you for your time!

**Comment by tabalinas** [comment]
If you `return false` from handler the event propagation will be stopped. And I'm wondering how it helps you to fix the issue with links in IE8? You handler even won't be fired, right? Or I'm getting it wrong?

**Comment by bboyz269** [comment]
Hi, you are correct. 

`return false` does not trigger my handler but it prevents "beforeunload" to be triggered as well. So it will fix my issue which is * confirm dialog is being prompted when click on the navigation anchor when using "beforeunload" event *.

To completely fix this issue, I think we must use something else instead of `$pagerContainer.emty();` to preserve all it's children event handlers. (like `$.hide()` or `$.detach()` then save elements to somewhere). That would take lots more effort and binding new event handlers to navigation anchor supporting IE 8 is not a common need in my opinion.

**Comment by tabalinas** [comment]
Really sure, if it's a good solution to `return false` because it will prevent other handlers from firing and event won't bubble. 
I guess, it's better for your particular case just attach handlers to links and call `e.preventDefault()` to avoid `beforeunload` firing.
Regarding `empty()`: I'm not sure how it influences on your case, since you have delegated handlers on the document?

**Comment by bboyz269** [comment]
Yeah, that's why it's so weird of IE8. I'll go with `e.preventDefault()` for now. Thanks.

**Comment by tabalinas** [comment]
You are welcome!
